tests/optim/test_oss.py

```diff
@@ -754,8 +802,29 @@ def closure_sharded(input_tensor=input_tensor):
                 loss_ddp, loss_sharded_optim
             ), f"Losses differ in between Pytorch optim and OSS\nworld size {world_size}"
 
+        # The model should be synchronized in between the ranks at construction time, check that
+        check_same_model_params()
+
+        # The models should stay the same in between ddp and sharded optimizer
+        for i in range(20):
+            check_step()
             check_same_model_params()
 
+        # Check that the checkpoints are compatible
+        # - get states
+        ddp_state_dict = ddp_optimizer.state_dict()
+        sharded_optimizer.consolidate_state_dict()
+        sharded_optim_state_dict = sharded_optimizer.state_dict()
+        sharded_optim_state_dict = sync_object_ranks(sharded_optim_state_dict, 0, device)
+
+        # - cross load the states
+        ddp_optimizer.load_state_dict(sharded_optim_state_dict)  # mixup on purpose !
```

Comment:
one side effect from this "elastic" change: the checkpoints are provably pytorch compatible

Reply:
this is amazing! I may add a test later on to check this works with model parallelism too

Reply:
@msbaines the states are saved on both pytorch and oss sides (defined with multiple groups) and cross loaded